mlir/include/mlir/IR/AffineExpr.h

```diff
@@ -96,6 +96,20 @@ class AffineExpr {
   /// constants, i.e., it does not involve dimensional identifiers.
   bool isSymbolicOrConstant() const;
 
+  /// Returns true if this expression is made out of only symbols.
+  bool isSymbolic() const;
+
+  /// Returns true if the expression is divisible by the given symbol.
```

Comment:
All of these should have been local/static methods and don't need to be exposed on the class.